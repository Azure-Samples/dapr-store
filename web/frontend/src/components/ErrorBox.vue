<!--
// ----------------------------------------------------------------------------
// Copyright (c) Ben Coleman, 2020
// Licensed under the MIT License.
//
// Dapr Store frontend - Reusable error alert box component
// ----------------------------------------------------------------------------
-->

<template>
  <b-alert v-if="error" show variant="warning">
    <h4>There was a problem ðŸ˜¥</h4>
    <div class="errmsg">
      {{ error }}
    </div>
  </b-alert>
</template>

<script>
export default {
  name: 'ErrorBox',

  props: {
    error: {
      validator: (prop) => typeof prop === 'string' || typeof prop === 'object' || prop === null,
      required: true
    }
  },
}
</script>

<style scoped>
.errmsg {
  color: #222;
  font-family: 'Consolas', 'Courier New', Courier, monospace;
}
</style>