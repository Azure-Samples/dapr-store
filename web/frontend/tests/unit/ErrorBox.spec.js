import { createLocalVue, shallowMount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
const localVue = createLocalVue()
localVue.use(BootstrapVue)

import ErrorBox from '@/components/ErrorBox.vue'

describe('ErrorBox.vue', () => {
  it('renders error message', () => {
    const errorMsg = 'This is an error'
    const wrapper = shallowMount(ErrorBox, { localVue,
      propsData: { error: errorMsg }
    })
    expect(wrapper).toMatchSnapshot()
  })
})