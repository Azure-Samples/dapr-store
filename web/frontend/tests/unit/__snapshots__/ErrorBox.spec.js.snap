// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox.vue renders error message 1`] = `
<b-alert-stub variant="warning" dismisslabel="Close" show="true">
  <h4>There was a problem ðŸ˜¥</h4>
  <div class="errmsg">
    This is an error
  </div>
</b-alert-stub>
`;
